### DISCUSSION TITLE
LILYGO TTV transparent screen as HUD?

### DISCUSSION THREAD

**Discussion by greymda** [alexanderlavrushko/BLE-HUD-navigation-ESP32#12] [Ideas]
i've stumble on the internet on this ESP32 model with.. transparent screen! link [here](https://www.aliexpress.com/item/1005001689823009.html) 

![lilygo](https://user-images.githubusercontent.com/9991596/125839658-dfa13db0-cc98-4cc0-8265-82f59d816ae8.jpeg)

as it's ESP32, do you think your code would work on this board? i see it as a dream come true, genuinely HUD for a car!

thank you for your time and work! and sorry for being annoying.

**Comment by alexanderlavrushko** [comment]
I expect that code which receives navigation data will work, but communication with **this type of display isn't implemented**, and not planned.

I think it would be good to reorganize the code for ESP32 similar way how it's done for nRF51 ([link](https://github.com/alexanderlavrushko/BLE-HUD-navigation-ESP32/blob/c8db940012b4af71dd5017e65d580cecf737ee72/nRF51-Arduino/BLEHUDNavi_nRF51/DataPresenter.h#L25)), so every ESP32-enthusiast can take HUDData and present it with any display they have. Also moving BLE communication code to a library would be an option, maybe some day...

**Reply by greymda** [reply]
got it, thanks for replying. 
i’m new to all this, but hopefully will understand how it’s working and what to change, lol

keep up the amazing work you’re doing!
